File: Competitive Programming 3/Section 1.3.3/10963 - The Swallowing Ground/Python/main.py
def main():
    t = int(input())
    for cs in range(0,t):
        input()
        if(cs>0):
            print()
        n = int(input())
        y1,y2 = map(int,input().split())
        dif = y1-y2
        ans = "yes"
        for _ in range(1,n):
            y1,y2=map(int,input().split())
            if(y1-y2!=dif):
                ans="no"
        print(ans)
main()

File: Competitive Programming 3/Section 1.3.3/10963 - The Swallowing Ground/Java/Main.java
import java.util.Scanner;

public class Main {
    static Scanner s = new Scanner(System.in);
    public static void main(String[] args){
        int t = Integer.parseInt(s.nextLine());
        for(int cs = 0 ; cs < t ; cs++){
            s.nextLine();
            if(cs>0)
                System.out.println();
            int n = Integer.parseInt(s.nextLine());
            String [] tk = s.nextLine().split(" ");
            int dif = Integer.parseInt(tk[0]) - Integer.parseInt(tk[1]);
            String ans = "yes";
            for(int i = 1 ; i < n ; i ++){
                tk = s.nextLine().split(" ");
                if(Integer.parseInt(tk[0]) - Integer.parseInt(tk[1])!=dif){
                    ans = "no";
                }
            }
            System.out.println(ans);
        }
    }
}